source/wp-content/themes/wporg-news-2021/blocks/post-type-name/index.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace WordPressdotorg\Theme\News_2021\Blocks\PostTypeName;
+
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+add_action( 'enqueue_block_assets', __NAMESPACE__ . '\register_block_type_js' );
+
+/**
+ * Renders the `wporg/post-type-name` block on the server.
+ *
+ * The block render the post type name. It also optionally renders it as a link.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string
+ */
+function render_block( $attributes, $content, $block ) {
+	if ( ! isset( $block->context['postId'] ) ) {
+		return '';
+	}
+
+	$post_id  = $block->context['postId'];
+	$post_type = get_post_type( $post_id );
+	$post_type_obj = get_post_type_object( $post_type );
+	$wrapper_attributes = get_block_wrapper_attributes();
+
+	if ( isset( $attributes['isLink'] ) && $attributes['isLink'] ) {
+		return sprintf(
+			'<a %1$s href="%2$s">%3$s</a>',
```

Comment:
Since you can pass in the `tagName`, styling via the tag name is not advised. Additionally, if we wanted tag consistency, as in always output a span, we would get:

```
<span><h1>Podcast</h1></span>
```
We could omit the `<span>` if the `tagName` is supplied but that may be more complicated than just not having it?